### ISSUE TITLE
SigLIP2 ImageNet test got 74% at ImageNet-1K

### ISSUE THREAD

**Issue by cjx96** [LAION-AI/CLIP_benchmark#135]
Hello, authors:

I have a misunderstanding phenomenon when I use the huggingface ckpt to evaluate.

First, I implement a huggingface ckpt loader as follows:

<code>
from transformers import AutoProcessor, AutoModel
class ProcessorWrapper:
    def __init__(self, processor):
        self.processor = processor 
    def __call__(self, images):
        if isinstance(images, list):
            return torch.stack([self.processor(images=img, return_tensors="pt")["pixel_values"][0] for img in images])
        else: 
            return self.processor(images=images, return_tensors="pt")["pixel_values"][0]
    def tokenizer(self, texts, **kwargs):
        kwargs.setdefault("padding", "max_length")
        kwargs.setdefault("truncation", True)
        kwargs.setdefault("max_length", 64)
        kwargs.setdefault("return_tensors", "pt")
        if isinstance(texts, str): 
            texts = [texts]
        return self.processor(text=texts, **kwargs)
class ModelWrapper:
    def __init__(self, model):
        self.model = model 
    def encode_image(self, image, **kwargs):
        return self.model.get_image_features(pixel_values=image)
    def encode_text(self, sample_input_ids, **kwargs):
        return self.model.get_text_features(input_ids=sample_input_ids["input_ids"])
    def eval(self,):
        return self.model.eval()
def load_hugginface(model_name: str = "useless", pretrained: str = "laion400m_e32", cache_dir: str = None, device="cpu"):
    model = AutoModel.from_pretrained(pretrained, trust_remote_code=True)
    model = model.to(device)
    processor = AutoProcessor.from_pretrained(pretrained, trust_remote_code=True)
    hugginface_processor = ProcessorWrapper(processor)
    hugginface_model = ModelWrapper(model)
    return hugginface_model, hugginface_processor, hugginface_processor.tokenizer

Next, utilizing the load method, I test the SigLIP and SigLIP2 ckpt from Huggingface. The SigLIP was right:

`{"dataset": "imagenet1k", "model": "SigLIP-So400M-Patch14-384", "pretrained": "/models/SigLIP-So400M-Patch14-384", "task": "zeroshot_classification", "metrics": {"acc1": 0.8308, "acc5": 0.97038, "mean_per_class_recall": 0.8310799999999999}, "language": "en"}`

**but the SigLIP2 was much lower than the original paper**
`
{"dataset": "imagenet1k", "model": "SigLIP-So400M-Patch14-384", "pretrained": "/models/siglip2-so400m-patch14-384", "task": "zeroshot_classification", "metrics": {"acc1": 0.74292, "acc5": 0.87556, "mean_per_class_recall": 0.74294}, "language": "en"}
`

However, the **open_clip load method** can reproduce the expected results for SigLiP2:
`
{"dataset": "imagenet1k", "model": "ViT-SO400M-16-SigLIP2-384", "pretrained": "webli", "task": "zeroshot_classification", "metrics": {"acc1": 0.84176, "acc5": 0.97454, "mean_per_class_recall": 0.8418199999999999}, "language": "en"}
`

I have checked the weights between "ViT-SO400M-16-SigLIP2-384" and "siglip2-so400m-patch14-384", **they are almost the same**, with a minor position embedding mismatch.

Is the model-forward different between the  "ViT-SO400M-16-SigLIP2-384" and "siglip2-so400m-patch14-384"? I have been struggling with this problem for several weeks, sad.

**Comment by cjx96** [comment]
<code>
from transformers import AutoProcessor, AutoModel
class ProcessorWrapper:
    def __init__(self, processor):
        self.processor = processor 
    def __call__(self, images):
        if isinstance(images, list):
            return torch.stack([self.processor(images=img, return_tensors="pt")["pixel_values"][0] for img in images])
        else: 
            return self.processor(images=images, return_tensors="pt")["pixel_values"][0]
    def tokenizer(self, texts, **kwargs):
        kwargs.setdefault("padding", "max_length")
        kwargs.setdefault("truncation", True)
        kwargs.setdefault("max_length", 64)
        kwargs.setdefault("return_tensors", "pt")
        if isinstance(texts, str): 
            texts = [texts]
        return self.processor(text=texts, **kwargs)
class ModelWrapper:
    def __init__(self, model):
        self.model = model 
    def encode_image(self, image, **kwargs):
        return self.model.get_image_features(pixel_values=image)
    def encode_text(self, sample_input_ids, **kwargs):
        return self.model.get_text_features(input_ids=sample_input_ids["input_ids"])
    def eval(self,):
        return self.model.eval()
def load_hugginface(model_name: str = "useless", pretrained: str = "laion400m_e32", cache_dir: str = None, device="cpu"):
    model = AutoModel.from_pretrained(pretrained, trust_remote_code=True)
    model = model.to(device)
    processor = AutoProcessor.from_pretrained(pretrained, trust_remote_code=True)
    hugginface_processor = ProcessorWrapper(processor)
    hugginface_model = ModelWrapper(model)
    return hugginface_model, hugginface_processor, hugginface_processor.tokenizer

**Comment by cjx96** [comment]
WHAT! THE！FUCK！It works for me. Thank you very much!!!!
